python/tank/authentication/login_dialog.py

```diff
@@ -325,22 +342,28 @@ def _link_activated(self, site):
                 self.ui.message, "Can't open '%s'." % forgot_password
             )
 
-    def _toggle_sso(self):
+    def _toggle_web(self):
         """
-        Sets up the dialog GUI according to the use of SSO or not.
+        Sets up the dialog GUI according to the use of web login or not.
         """
         # We only update the GUI if there was a change between to mode we
         # are showing and what was detected on the potential target site.
-        if self._use_sso != self._query_task.sso_enabled:
-            self._use_sso = not self._use_sso
-            if self._use_sso:
-                self.ui.message.setText("Sign in using your Single Sign-On (SSO) Account.")
+        use_web = self._query_task.sso_enabled or self._query_task.autodesk_identity_enabled
+
+        # If we have full support for Web-based login, or if we enable it in our
+        # environment, use the Unified Login Flow for all authentication modes.
+        if get_shotgun_authenticator_support_web_login():
+            use_web = use_web or self._query_task.unified_login_flow_enabled
+        if self._use_web != use_web:
+            self._use_web = not self._use_web
```

Comment:
any chance you could add some more comments here? I find this a little confusing!